quickwit/quickwit-ingest/src/ingest_v2/workbench.rs

```diff
@@ -51,6 +52,28 @@ pub(super) struct IngestWorkbench {
     pub unavailable_leaders: HashSet<NodeId>,
 }
 
+pub(super) fn pending_subrequests(
+    subworkbenches: &BTreeMap<SubrequestId, IngestSubworkbench>,
+) -> impl Iterator<Item = &IngestSubrequest> {
+    let pending_subrequests_iterator = subworkbenches.values().filter_map(|subworbench| {
+        if subworbench.is_pending() {
+            Some(&subworbench.subrequest)
+        } else {
+            None
+        }
+    });
+    // We sort the output in tests, to make the output deterministic.
+    #[cfg(test)]
+    {
+        use itertools::Itertools;
+        pending_subrequests_iterator.sorted_by_key(|subrequest| subrequest.subrequest_id)
+    }
```

Comment:
I'm not sure this is required (or rather, don't understand why it would be), the iterator should already be sorted because of the BTreeMap.

fwiw i ran the tests a few time without that, and didn't get any error related to this

Reply:
oh yes you are right!